Demo/Demo/SwiftUIComponents/PayPalWebPayments/PayPalWebPaymentsView.swift

```diff
@@ -0,0 +1,25 @@
+import SwiftUI
+
+struct PayPalWebPaymentsView: View {
+
+    @StateObject var payPalWebViewModel = PayPalWebViewModel()
+
+    var body: some View {
+        ScrollView {
+            VStack(spacing: 16) {
+                PayPalWebCreateOrderView(payPalWebViewModel: payPalWebViewModel)
+                if payPalWebViewModel.order != nil {
+                    PayPalWebResultView(payPalWebViewModel: payPalWebViewModel, status: .created)
+                    NavigationLink {
```

Comment:
If there is an error in creating order, the error view would not be displayed if we check for nil value for order.

Reply:
What do you think of this solution? It was the quickest fix from what you have now. Instead of single state,
I have createOrderState, approveOrderState, authorizeCaptureOrderState of CurrentState type initialized to .idle.
I check for these states' success on the buttons to next step. In PayPalWebResultView, I toggle the state based on status. 

struct PayPalWebResultView: View {

    @ObservedObject var payPalWebViewModel: PayPalWebViewModel

   ```
 var status: OrderStatus
    var state: CurrentState {
        switch status {
        case .approved:
            return payPalWebViewModel.approveOrderState
        case .created:
            return payPalWebViewModel.createOrderState
        case .completed:
            return payPalWebViewModel.authorizeCpatureOrderState
        }
    }

    var body: some View {
        switch state {
        case .idle, .loading:
            EmptyView()
```
Of course, I had to update the "update" functions in the viewModel.
It seemed to work. I can put up a branch if you like.

Reply:
Let me check but I do recall seeing the error view in main.

Reply:
I've gone ahead and fixed the bug that existed on main with ErrorView, additionally added different results which I think is a _slightly_ better way then handing separate states in the latest 2 commits. Personally I feel that this amount of switching of handling views points out a flaw in our demo app. While this PRs intent was to simplify the views I think we can take a second pass at further cleaning up the demo app for this feature before moving on to the next one. 

Specifically we should have this all on 1 view (similar to card) with 1 result that is appended to as we have more data (vs the separate status views that exist on main/in this PR). I think another couple of refactors that we can consider is a picker view for the buttons (similar to auth/capture when creating an order) as well as 1 button that changes as we progress from create order > button view > transaction view. Additionally we don't have a way to reset PayPal web so running the demo multiple times become difficult as we never reset the views.

I think this PR can/should be the first of a few progressive refactors before we move onto other feature! 🚀

Reply:
I did have results appended in Card Vault without purchase and that made it easier to have "reset" button. 
With different states for each API results, that makes things easier.

Reply:
> The error I was pointing out was the fact that you guarded the result view with success or value from API and that prevented errorViews from being surfaced.

Yep this was the same on main as I said ([code here](https://github.com/paypal/paypal-ios/blob/ae1adc3788995995adb677f39826d49bbdd3ffeb/Demo/Demo/SwiftUIComponents/PayPalWebViews/PayPalWebView.swift#L15)) - we never handled the else so if an order wasn't created we just console logged the error but never displayed the error view.

> With different states for each API results, that makes things easier.

We should be able to remove the need for different states entirely with moving to a singular success view that's appended to as we get additional data. This would be more in line with the Braintree demo app as well (we actually display way less data there and if we need additional information for troubleshooting rely on breakpoints) - I think there is a good middle ground between the two though.

Reply:
ooh, that was an oversight just for the order creation result. I meant to do what I did in TransactionView which was to guard only the next step views. Thank you for correcting that as well.

Reply:
No problem! It was the same on the final view as well (we only previously handled if auth/capture was not nil - [here](sdks-messages@paypal.com) - can be tested by hardcoding the string "orderID" in the auth/capture call).

Reply:
Yes, my intention was what I showed above in PayPalTransactionView.